### PR TITLE
Singularity and Narsie Refactor

### PR DESCRIPTION
**Pull request by Fox-McCloud**
Large Singularity Overhaul

This PR does three things (though it has a number of small things connected to it)

- Ports over TG's OOP'd Singularity
- Ports over TG's own implementation of VG's Narsie
- Ports over TG's proposed PR for a super singulo (https://github.com/tgstation/-tg-station/pull/9847)
 - More than happy to revert this from the PR, if necessary; it is an exceedingly rare thing, especially with Paradise's "responsible antags" (and has a huge sprite that will almost never be seen).

Singularity Changes
- Overall, the singularity is far more OOP'd
- Singularity is now an object as opposed to machine
- New features are that it has more unique effects based on its size
 - Size 4 and higher (right before it goes critical) will start making tiles into plating, throwing items at itself, pulling things out of your hands, and ripping the plating off of walls.
 - If/when we refactor floors, it'll rip up floor tiles, too.
- if the singularity consumes the supermatter, it can attain level 6, which has even greater range and combusts mobs in range. More info here: https://github.com/tgstation/-tg-station/pull/9847

Narsie
- Narsie now converts floors and walls into cult versions of floors/walls instead of eating everything
- Touching Narsie will now gib you and turn you into a harvester
- Ghosts who click on Narsie will be turned into a harvester
 - Overall lag generated by new Narsie is nearly non-existent and it offers players a far more interesting and enjoyable ~15 minutes of the round with less lag and more interaction/gameplay

Related
- Cleans up the singularity beacon
- Adds in harvesters
 - These are a type of cult construct that is very weak; their primary purpose is to act as a pest and drag unconscious people to Narsie for conversion.

Code wise, would appreciate this be heavily reviewed.

### CODE REVIEW

File: code/modules/power/singularity/singularity.dm [hunk 1]
```diff
@@ -1,27 +1,15 @@
...
+/obj/singularity
+	name = "gravitational singularity"
+	desc = "A gravitational singularity."
 	icon = 'icons/obj/singularity.dmi'
 	icon_state = "singularity_s1"
 	anchored = 1
 	density = 1
 	layer = 6
-	light_range = 6
+	luminosity = 6
```
Comment (tigercat2000) [inline code/modules/power/singularity/singularity.dm]: That'll break things, it needs to be light_range.

**Comment by fj45** [comment]
Does the sing change cause any lag?

**Comment by Spacemanspark** [comment]
:-1: to the singularity NO.
NO.
NO GOD NO.
PLEASE NO.
Everything else is fine, and Nar-sie changes are :+1: :x: :100:
Just. No. To. The. Singularity. Changes.

**Comment by Fox-McCloud** [comment]
The throwing of items towards the singularity only happens when it hits stage 4 and 5 (well, technically 6, too, but that's a special stage); the range at which is pulls mobs is the same range at which it pulls+throws items...meaning you're going to have to be 1 tile closer to the singularity than the items getting thrown around....anddd that means you're already screwed, at that point.

**Comment by KasparoVy** [comment]
Narsie changes are amazing, just reading it made me excited

singulo sounds legit as hell too

**Comment by Krausus** [comment]
Oh, you ported over that stage 6 singularity.

The one with the goddamn 2.2MB sprite that will basically never be seen outside of adminbus or someone going through a lot of trouble to grief as hard as possible.

No. :-1:

**Comment by ZomgPonies** [comment]
Oh, so we're deleting sprites to add in more giant sprites now? Heh.

**Comment by Mijhead** [comment]
:+1: , Singulo stuff, especially the lvl 6 one, sounds cool as fuck.

**Comment by Fox-McCloud** [comment]
Would be willing to axe the level 6 if it doesn't prove popular enough, as codebloat is a major concern of mine.

**Comment by monkeysfist101** [comment]
The Nar'sie change might be okay, but ❌ to the singularity change.

**Comment by Earthdivine** [comment]
I don't understand why everyone's going lol nope to the Singlarity. It makes me more lethal....at the stages it will kill you anywas. Stage 4 is bearly ever even seen outside the pen and if your at that range of a 5 your dead. All I can see this doing it pulling in Space Pod Pilots who want to stare at it.

**Comment by Spacemanspark** [comment]
@Earthdivine As someone who's experienced it in actual gameplay, if it hadn't ripped shit up and tossed it at me, I would have made it away safely. In fact, a few times the floor tiles have killed me, then the singularity moved away. It's pretty much just a giant 'fuck you' to the face. And it makes griefing the singularity just that much more satisfying to those who will pursue it.

**Comment by Earthdivine** [comment]
That said, it would have hardly mattered if you had died though, as a Singlo escaping is usually grounds for the Shuttle, or just gets flat out deleted by the admins if it's warrented. All this change is making a cool thing even cooler.

That said, if it really is such a concern, would it be better to finally actually add in a way to deal with an escaped Singlo that is NOT admin intervention.

**Comment by Fox-McCloud** [comment]
@Earthdivine 

That's actually already included in this PR; explosions will now destroy a level 2 singularity 100% of the time (instead of being RNG 25% chance), and other explosions will now *reduce* the amount of energy the singularity has, as opposed to increasing it.

Naturally, Bags of Holding are still the best way to kill a singulo.

**Comment by IAmBigCoat** [comment]
My only gripe is the line in the PR you pulled from.
"Also, I know this code is probably not optimized well."

Otherwise, it's always nice having more things for ghosts to do. :+1:

**Comment by Regen1** [comment]
I love it all, especially the singularity.
:+1:

**Comment by Necaladun** [comment]
This all sounds awesome. I love the sprite. I love narsie being different to a sing...I love it all.

**Comment by ZomgPonies** [comment]
:+1:

**Pull request merged**